src/directions/main.ts

```diff
@@ -919,21 +971,34 @@ export default class MapLibreGlDirections extends MapLibreGlDirectionsEvented {
    * Replaces all the waypoints with the specified ones and re-fetches the routes.
    *
    * @param waypoints The coordinates at which the waypoints should be added
+   * @param profiles Profiles for fetching directions between waypoints
    * @return Resolved after the routing request has finished
    */
-  async setWaypoints(waypoints: [number, number][]) {
+  async setWaypoints(waypoints: [number, number][], profiles: string[] = []) {
     this.abortController?.abort();
 
-    this._waypoints = waypoints.map((waypoint, index) => {
-      return this.buildPoint(
-        waypoint,
-        "WAYPOINT",
-        this.configuration.bearings
-          ? {
-              bearing: this.waypointsBearings[index],
-            }
-          : undefined,
-      );
+    this.profiles = profiles.slice(0, waypoints.length - 1);
+    if (this.profiles.length === 0) {
+      this.profiles.push(this.configuration.profile);
+    }
```

Comment:
Please, add comments here for the future readers, explaining the logic of what that code's doing.

Reply:
Actualy, would also be nice to add some comments to the `fetch`/`fetchDirections` parts as well, but, I guess, that's optional, since (if I'm not mistaken) there were none before your contribution. But I'd appreciate them anyway, or otherwise there's a risk that future readers won't understand a thing that's going on there (it's already kinda hard to get the grasp of).

Reply:
Described some crucial logic in the comments